lib/AST/Decl.cpp

```diff
@@ -2259,9 +2259,21 @@ ValueDecl::getFormalAccessScope(const DeclContext *useDC,
   llvm_unreachable("unknown access level");
 }
 
-void ValueDecl::copyFormalAccessFrom(ValueDecl *source) {
+void ValueDecl::copyFormalAccessFrom(const ValueDecl *source,
+                                     bool sourceIsParentContext) {
```

Comment:
Do you need sourceIsParentContext, or can you just check if the source's parent DeclContext is a SourceFile, and if so, turn Private to FilePrivate?

Reply:
It's relevant for any type, unfortunately, not just those at the top level. If we wanted to do it without the flag we could check if `source` and `this` are related in some way, but that seems more complicated for something every call site already knows.